botocore/endpoint.py

```diff
@@ -152,16 +130,13 @@ def _create_request_object(self, request_dict):
                              headers=headers)
         return request
 
-    def prepare_request(self, request, signer):
-        if signer is not None:
-            with self._lock:
```

Comment:
The develop branch should be thread safe, as auth_path is now set on the request instead of the auth class.  I'd still suggest double checking that this is the case though.